tests/jdom/button.spec.ts

```diff
@@ -0,0 +1,144 @@
+// Test the ButtonServer, intended as a test for the unit test framework.
+// Yes, we're meta-testing!
+
+import { suite, test } from "mocha"
+import { ButtonEvent } from "../../src/jdom/constants"
+import { withBus, createServices, nextEventFrom } from "./tester"
+import { assert } from "../../src/jdom/utils"
+import ButtonServer from "../../src/servers/buttonserver"
+
+suite("button server", () => {
+    test("fires edge events after changing state", async function () {
+        const buttonServer = new ButtonServer("button", false)
+
+        await withBus(async bus => {
+            const { button } = await createServices(bus, {
+                button: buttonServer,
+            })
+
+            buttonServer.down()
+            assert(
```

Comment:
can we use a js assertion library here as well? typically you'd use something like "assert.equal". No need to write our own library, the point here would be to also use an off-the-shelve assertion lib.

Reply:
chai is very popular. https://www.chaijs.com/